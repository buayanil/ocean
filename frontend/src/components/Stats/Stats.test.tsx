import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Stats } from "./Stats";

describe("Stats Component", () => {
    it("renders the stat name and value correctly", () => {
        // Arrange
        const mockName = "Total Users";
        const mockValue = "12345";

        // Act
        render(<Stats name={mockName} value={mockValue} />);

        // Assert
        expect(screen.getByText(mockName)).toBeInTheDocument(); // Check if the name is displayed
        expect(screen.getByText(mockValue)).toBeInTheDocument(); // Check if the value is displayed
    });

    it("applies correct classes for styling", () => {
        // Arrange
        const mockName = "Revenue";
        const mockValue = "$1,000,000";

        // Act
        const { container } = render(<Stats name={mockName} value={mockValue} />);

        // Assert
        const statElement = container.querySelector(".shadow.rounded-lg");
        expect(statElement).toBeInTheDocument();
        expect(statElement).toHaveClass("px-4", "py-5", "bg-white", "shadow", "rounded-lg");
    });

    it("renders a snapshot of the component", () => {
        // Arrange
        const mockName = "Active Users";
        const mockValue = "234";

        // Act
        const { asFragment } = render(<Stats name={mockName} value={mockValue} />);

        // Assert
        expect(asFragment()).toMatchSnapshot();
    });
});
