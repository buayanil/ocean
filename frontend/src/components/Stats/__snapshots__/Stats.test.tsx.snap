// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component renders a snapshot of the component 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-5 bg-white shadow rounded-lg overflow-hidden sm:p-6"
  >
    <dt
      class="text-sm font-medium text-gray-500 truncate"
    >
      Active Users
    </dt>
    <dd
      class="mt-1 text-3xl font-semibold text-gray-900"
    >
      234
    </dd>
  </div>
</DocumentFragment>
`;
